app/models/discussion_reader.rb

```diff
@@ -86,10 +86,21 @@ def viewed!(age_of_last_read_item = nil)
            last_read_at: age_of_last_read_item)
   end
 
+  def reset_items_count!
+    self.read_items_count = count_read_items(last_read_at)
+    self.last_read_sequence_id = lookup_last_read_item.try(:sequence_id) || 0
+    self.save!(validate: false)
+  end
+
+  def reset_comments_count!
+    update_attribute(:read_comments_count, count_read_comments(last_read_at))
+  end
+
   def reset_counts!
-    self.read_comments_count = count_read_comments(last_read_at)
     self.read_items_count = count_read_items(last_read_at)
-    self.save!
+    self.read_comments_count = count_read_comments(last_read_at)
+    self.last_read_sequence_id = lookup_last_read_item.try(:sequence_id) || 0
```

Comment:
Same thing here, I'd at least consider

```
discussion.items.where('created_at < ?', last_read_at).maximum(:sequence_id)
```

Since I can't see a place where we actually care what the last item is, we just care what the last sequence id is.